src/synthesizer/particle/blackholes.py

```diff
@@ -672,6 +675,13 @@ def generate_particle_lnu(
         spec = np.zeros((self.nbh, masked_spec.shape[-1]))
         spec[mask] = masked_spec
 
+        # Apply the wavelength mask if provided
+        spec[:, ~lam_mask] = 0.0
+
+        # TODO: Wavelength masking could be done before the C function call to
```

Comment:
HA, like you say here 😂 but doesn't require any grid modifications. We can just apply the lam mask in the argument preparation method.